compiler/env/Region.cpp

```diff
@@ -90,6 +90,15 @@ Region::allocate(size_t const size, void *hint)
    return _currentSegment.get().allocate(roundedSize);
    }
 
+void *
+Region::reallocate(size_t newsize, void *pointer, size_t size)
+{
+   void * npointer = allocate(newsize);
+   memcpy(npointer, pointer, newsize < size ? newsize : size);
+   deallocate(pointer, size);
```

Comment:
I agree, it would be better if the `Region` provides some way to free the memory and maintain the free list internally. I think the issue you have opened #1995 is ideal to carry out that work.